pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -41,27 +45,86 @@ var log = logf.Log.WithName("DeploymentReconciler")
-	scheme       *runtime.Scheme
-	Deployment   *appsv1.Deployment
-	componentExt *v1beta1.ComponentExtensionSpec
+	client         kclient.Client
+	scheme         *runtime.Scheme
+	DeploymentList []*appsv1.Deployment
+	componentExt   *v1beta1.ComponentExtensionSpec
 }
 
 func NewDeploymentReconciler(client kclient.Client,
 	scheme *runtime.Scheme,
 	componentMeta metav1.ObjectMeta,
+	workerComponentMeta metav1.ObjectMeta,
 	componentExt *v1beta1.ComponentExtensionSpec,
-	podSpec *corev1.PodSpec) *DeploymentReconciler {
+	podSpec *corev1.PodSpec, workerPodSpec *corev1.PodSpec) *DeploymentReconciler {
 	return &DeploymentReconciler{
-		client:       client,
-		scheme:       scheme,
-		Deployment:   createRawDeployment(componentMeta, componentExt, podSpec),
-		componentExt: componentExt,
+		client:         client,
+		scheme:         scheme,
+		DeploymentList: createRawDeployment(componentMeta, workerComponentMeta, componentExt, podSpec, workerPodSpec),
+		componentExt:   componentExt,
 	}
 }
+func createRawDeployment(componentMeta metav1.ObjectMeta, workerComponentMeta metav1.ObjectMeta,
+	componentExt *v1beta1.ComponentExtensionSpec,
+	podSpec *corev1.PodSpec, workerPodSpec *corev1.PodSpec) []*appsv1.Deployment {
+	var deploymentList []*appsv1.Deployment
+	var workerNodeReplicas int32
+	var tensorParallelSize string
+	multiNodeEnabled := false
+
+	if workerComponentMeta.Name != "" {
+		multiNodeEnabled = true
+
+		for _, container := range podSpec.Containers {
+			if container.Name == constants.InferenceServiceContainerName {
+				if value, exists := utils.GetEnvVarValue(container.Env, constants.PipelineParallelSizeEnvName); exists {
+					if parsedValue, err := strconv.Atoi(value); err == nil {
+						// Set pipelineParallelSize to workerNodeSize + 1 (head)
+						workerNodeReplicas = int32(parsedValue - 1) // nolint  #nosec G109
+					} else {
+						log.Error(err, "Failed to convert pipelineParallelSize to int")
+					}
+				} else {
+					log.Info(fmt.Sprintf("PIPELINE_PARALLEL_SIZE is not set in the container's environment(%s)", constants.InferenceServiceContainerName))
+				}
+				break
+			}
+		}
+	}
+
+	defaultDeployment := createRawDefaultDeployment(componentMeta, componentExt, podSpec)
+	if multiNodeEnabled {
+		// Use defaut value(1) if tensor-parallel-size is not set (gpu count)
+		tensorParallelSize = constants.DefaultTensorParallelSize
+
+		for _, container := range podSpec.Containers {
+			if container.Name == constants.InferenceServiceContainerName {
+				if value, exists := utils.GetEnvVarValue(container.Env, constants.TensorParallelSizeEnvName); exists {
+					// Use the environment variable value
+					tensorParallelSize = value
+				}
+				break
+			}
+		}
+		// Update GPU resource of default podSpec
+		addGPUResourceToDeployment(defaultDeployment, constants.InferenceServiceContainerName, tensorParallelSize)
+	}
+	deploymentList = append(deploymentList, defaultDeployment)
+
+	// Adds workerNode deployment
+	if multiNodeEnabled {
```

Comment:
No, this was structured this way because it was necessary for the logic. It was designed with the assumption that the default deployment is always the first item in the list.
If you read the logic, when multinode enabled, the default deployment need to be updated so it needs another `if`